### PR TITLE
test: add test cases for RuntimeDirectoryPreserve=yes

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: test/test-execute/exec-dynamicuser-runtimedirectory1.service [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=Test for RuntimeDirectory with RuntimeDirectoryPreserve=yes and DynamicUser=yes
+
+[Service]
+ExecStart=/bin/sh -x -c 'test -d %t/test-exec_runtimedirectorypreserve'
+ExecStart=/bin/sh -x -c 'test "$$RUNTIME_DIRECTORY" = "%t/test-exec_runtimedirectorypreserve"'
+ExecStart=/bin/sh -x -c 'touch $$RUNTIME_DIRECTORY/test'
```
Comment (keszybz) [inline test/test-execute/exec-dynamicuser-runtimedirectory1.service:7]: Not that it matter a lot, but we don't need `sh` in most cases. It is not wrong, but it'd be slightly simpler and more efficient to call `test` and `touch` directly in cases where envvars don't need to be expanded.
```
ExecStart=test -d %t/test-exec_runtimedirectorypreserve
ExecStart=sh -x -c 'test "$$RUNTIME_DIRECTORY" = "%t/test-exec_runtimedirectorypreserve"'
ExecStart=touch "%t/test-exec_runtimedirectorypreserve/test"
```

**Comment by keszybz** [comment]
@yuwata update?

**Comment by yuwata** [comment]
Still I have not found why the test fails... We need to investigate more.

**Comment by yuwata** [comment]
@keszybz I do not know which commit fixes the issue, but now the test passes. PTAL.

**Comment by keszybz** [comment]
This conflicts with #17079 a bit, but I'll need to push another version there anyway.

**Pull request merged**